pkg/cmd/scaffold.go

```diff
@@ -250,11 +252,16 @@ func scaffold(opts ScaffoldOptions) ([]byte, error) {
 	if err := validateFlags(opts); err != nil {
 		return nil, err
 	}
-
 	name, err := getNameFromImageReference(opts.from)
 	if err != nil {
 		return nil, err
 	}
+	if len(opts.name) > 0 {
+		if !validateName(opts.name) {
```

Comment:
I am wondering if we should add this validation to the name parsed from docker image ref as well.

Reply:
Hmm 🤔 In that case, I would move everything into the `getNameFromImageReference` function, rename it to `getSpinAppName` to streamline the code

Reply:
I think that should be good. My only reason for that was to keep the validation for `name` consistent irrespective of source of that string.

Reply:
@rajatjindal I've updated the PR according to the discussion we had here. As part of that, I've also added the `name` flag to the `deploy` sub command